game/src/main/kotlin/world/gregs/voidps/world/activity/skill/fletching/Bolts.kts

```diff
@@ -0,0 +1,48 @@
+package world.gregs.voidps.world.activity.skill.fletching
+
+import world.gregs.voidps.engine.client.message
+import world.gregs.voidps.engine.client.ui.interact.itemOnItem
+import world.gregs.voidps.engine.data.definition.data.FletchBolts
+import world.gregs.voidps.engine.entity.character.player.chat.ChatType
+import world.gregs.voidps.engine.entity.character.player.skill.Skill
+import world.gregs.voidps.engine.entity.character.player.skill.level.Level.has
+import world.gregs.voidps.engine.entity.item.Item
+import world.gregs.voidps.engine.inv.inventory
+import world.gregs.voidps.engine.inv.transact.operation.AddItem.add
+import world.gregs.voidps.engine.inv.transact.operation.RemoveItem.remove
+
+itemOnItem("feather", "*_bolts_unf") {
+    val boltUsed = if (toItem.boltsUnf) toItem else fromItem
+    val bolts: FletchBolts = boltUsed.def.getOrNull("fletch_bolts") ?: return@itemOnItem
+
+    if (!it.has(Skill.Fletching, bolts.level, true)) {
+        return@itemOnItem
+    }
+
+    val currentFeathers = it.inventory.count("feather")
+    val currentBoltUnf = it.inventory.count(boltUsed.id)
+
+    val actualAmount = minOf(currentFeathers, currentBoltUnf, 10)
+
+    if (actualAmount < 1) {
+        return@itemOnItem
+    }
```

Comment:
None of these classes have messages to tell the player they don't have enough X item. Is that correct?